### PR TITLE
Feature/zf3compat

### PR DESCRIPTION
**Pull request by thomasvargiu**
ZF3 compatibility

### CODE REVIEW

**Comment by MichaelGooden** [comment]
Can you resubmit this without all the PHPCS fixes? It is hard to review what has changed exactly when every second line has some CS type change.

(I am a fan of fixing the CS, but that should be in its own PR.)

**Comment by thomasvargiu** [comment]
@MichaelGooden done!

File: src/Mapper/AbstractMapper.php [hunk 1]
```diff
@@ -134,11 +135,12 @@ public function exchangeOAuth2Array(array $array)
                 case 'collection':
                     $oAuth2String = array();
                     $fieldValues = explode(' ', $value);
-
+                    /** @var QueryBuilder $queryBuilder */
                     $queryBuilder = $this->getObjectManager()->createQueryBuilder();
                     $queryBuilder->select('row')
-                    ->from($this->getConfig()->mapping->$key->entity, 'row')
-                    ->andwhere(
+                    ->from($this->getConfig()->mapping->$key->entity, 'row');
```
Comment (TomHAnderson) [inline src/Mapper/AbstractMapper.php]: Tab in fluent interfaces to indent from first call.
```
$queryBuilder->select('row')
    ->from(...
```

File: src/Query/OAuth2ServerInitializer.php [hunk 2]
```diff
@@ -2,18 +2,58 @@
...
+
+    /**
+     * @param ContainerInterface|ServiceLocatorInterface $container
+     * @param mixed $instance
+     * @return OAuth2ServerInterface
+     * @throws \InvalidArgumentException
+     */
+    protected function internalInitialize($container, $instance)
+    {
+        if (!$instance instanceof OAuth2ServerInterface) {
```
Comment (TomHAnderson) [inline src/Query/OAuth2ServerInitializer.php]:
```
(! $instance
```

**Comment by thomasvargiu** [comment]
@TomHAnderson I've done the the changes.

In the future, you should add a `.php_cs` configuration file with your specifications. ;)

**Comment by TomHAnderson** [comment]
https://github.com/API-Skeletons/zf-oauth2-doctrine/releases/tag/2.0.0

**Comment by MichaelGooden** [comment]
Hi Tom,

Considering that this is a wholly backwards compatible change, maybe you should release this as 1.1.0 instead/as well?

If people are using this project using the recommended semantic version approach, e.g. ^1.0.0. and they wish to migrate their project to using zend-mvc v3, by not having the ZF3 compatible code in the 1.x branch it will block their upgrade, meanwhile there is actually no change needed from them.

**Comment by thomasvargiu** [comment]
I agree with 1.1.0. It's fully compatible with both versions, no BC breaks. I personally used my fork to migrate a big application from ZF2 to ZF3.

Without both compatibility it's hard to migrate a project.

**Comment by TomHAnderson** [comment]
OK I've tagged at 1.1.0
Thanks for the input.
https://github.com/API-Skeletons/zf-oauth2-doctrine/releases/tag/1.1.0

**Comment by MichaelGooden** [comment]
Hmm, now of course composer is complaining that the version I am using (1.1.0) is out of date :D

**Comment by TomHAnderson** [comment]
I've removed 2.0.0 from packagist

**Comment by MichaelGooden** [comment]
Cool.

Just remember for if/when there is an actual version 2 of this library, the very first version must be 2.0.1 or 2.1.0, due to the never ever ever reuse version numbers rule.

**Comment by MichaelGooden** [comment]
Also, don't forget to delete the actual tag in the Git repository.

**Comment by thomasvargiu** [comment]
It's better to never delete a tag. What about to create a branch "2.0" and start develop a version only for ZF3 and PHP >= 7.0?

**Comment by TomHAnderson** [comment]
No, Yes.  (that's a valid answer in California).  The 2.0.0 branch was my mistake not seeing backwards compatibility.  A ZF3/7.0 only version is a fine idea and I will tag a bc release when we have 7.0 specific code.  In the mean time the 2.0.0 tag has been removed from api-skeletons/zf-oauth2-doctrine and I'm happy with where the project is today.

**Pull request merged**